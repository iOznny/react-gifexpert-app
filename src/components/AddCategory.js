import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    // Inicio del input
    const [inputValue, setInputValue] = useState('');

    // Cambiar valor del input
    const handleInputChange = (e) => {        
        setInputValue(e.target.value);
        console.log('handleInputChange llamado');
    };

    // Prevenir posteo de form y agregar nuevo valor al arreglo
    const handleSubmit = (e) => {                
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('');
        }
    }

    return (  
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue } 
                onChange={ handleInputChange }
            />
        </form>
    )
}

// Requerir el prop-types
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
