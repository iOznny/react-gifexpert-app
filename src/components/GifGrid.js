import React from 'react';
import PropTypes from 'prop-types';

// Hooks
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
    
    return (
        <>    
            <h5>{ category }</h5>    
            { loading && <p className="animate__animated animate__flash">Cargando...</p> }

            <div className="card-grid">
                {
                    images.map(img => (<GifGridItem key={ img.id } { ...img } />))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}