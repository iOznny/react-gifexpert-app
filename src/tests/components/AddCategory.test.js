import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Testing AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    beforeEach(() => {
        // Limpiar Mocks.
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });

    test('Show successful', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('You must change the text box', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        
        input.simulate('change', { target: { value }  }); 
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('Should not pass the information with submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Set Categories Clear text box', () => {

        const value = 'Hola Mundo';

        // Simular el inputchange
        wrapper.find('input').simulate('change', { target: { value } });

        // Simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // setCategories debe de haber llamado
        expect(setCategories).toHaveBeenCalled();        
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        // el valor del input debe estar ''
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});