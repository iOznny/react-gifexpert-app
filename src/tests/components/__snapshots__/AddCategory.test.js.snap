// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory Show successful 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
